src/main/scala/viper/gobra/translator/encodings/defaults/DefaultMethodEncoding.scala

```diff
@@ -74,10 +101,27 @@ class DefaultMethodEncoding extends Encoding {
     val vResultPosts = x.results.flatMap(ctx.varPostcondition)
     val vResultInit = cl.seqns(x.results map ctx.initialization)
 
+    //Depending if a private spec exists, we verify the public or private specification
+    val (vPres, vPosts, vMeasures) = if (x.privateSpec.isEmpty) {
+      (vArgPres ++ x.pres.map(ctx.precondition),
+       vResultPosts ++ x.posts.map(ctx.postcondition),
+       x.terminationMeasures.map(e => pure(ctx.assertion(e))(ctx)))
+    } else {
+      val spec = x.privateSpec.getOrElse(null) //not empty
+
+      //Generates a method to prove private entailment
+      privateProof.privateProofFunction(x)(ctx)
+
+      (vArgPres ++ spec.pres.map(ctx.precondition),
+       vResultPosts ++ spec.posts.map(ctx.postcondition),
+       spec.terminationMeasures.map(e => pure(ctx.assertion(e))(ctx)))
+      
+    }
```

Comment:
That does not work, because `x` has class `in.Method` and `x.privateSpec.get` has class `in.PrivateSpec`, so the variable `z` would have type `Any`. I would need to pattern match to get the class back to access `z.pres`, `z.posts`, and `z.terminationMeasures`. Therefore, the result would be the same but just with additional pattern matching.

Reply:
I see, makes sense.